app/services/app_store_token_provider.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class AppStoreTokenProvider
+  include Singleton
+
+  def initialize
+    oauth_client
+  end
+
+  def oauth_client
+    tenant_id = ENV.fetch('APP_STORE_TENANT_ID', nil)
+    @oauth_client ||= OAuth2::Client.new(
+      client_id,
+      ENV.fetch('APP_STORE_CLIENT_SECRET', nil),
+      token_url: "https://login.microsoftonline.com/#{tenant_id}/oauth2/v2.0/token"
+    )
```

Comment:
**nit** because we are memoizing, the tenant_id can/should be extracted to its own method and called here

```ruby
private 
...
def tenant_id
  ENV.fetch('APP_STORE_TENANT_ID', nil)
end
```